### ISSUE TITLE
UI freeze with MAM-enabled account and big roster

### ISSUE THREAD

**Issue by vitalyster** [ForNeVeR/kaiwa#69]
Using account with 150+ contacts takes 30+ seconds to login and UI freeze
Due to broken by design(tm) MAM XEP, we need to send 150+ MAM queries and process 150+ * N forwarded messages. We need to not freeze UI at least

**Comment by the-solipsist** [comment]
Can't the MAM requests be made *when* that particular contact is selected, rather than when Kaiwa merely opens?

**Comment by stevenroose** [comment]
@the-solipsist makes sense. Also, can't conversations be caches somewhat in the browser? Using IndexedDB?
